### DISCUSSION TITLE
Zero Consumers for Quroum Queues in RMQ Management Console

### DISCUSSION THREAD

**Discussion by dfcarver** [rabbitmq/rabbitmq-server#4008] [Questions]
We are trying to create Quorum queues and move off of classic queues.   We have noticed that there are ‘0’ consumers associated to new Quorum queue. Is this the expected behavior for Quorum queues?  For classic queues, the count is 9.  
![image001](https://user-images.githubusercontent.com/12265049/150142251-e83d22c2-6a06-4ebf-b81c-07afa53ab8c0.png)

We have the clients and these are old queues recreated as quorum queues.

**Comment by lukebakken** [comment]
We need to know the following:
* RabbitMQ version
* Erlang version
* Operating system

> We have noticed that there are ‘0’ consumers associated to new Quorum queue

That's probably because there are `0` consumers.

> We have the clients and these are old queues recreated as quorum queues.

If you can reproduce this issue reliably, share a script / code to do so. That will save a lot of time and back-and-forth here.

**Reply by dfcarver** [reply]
3 node cluster.
Total 6 queues are there and each queue has its own virtual host.
OS : Windows Server 2016, hosted on VMware.
Erlang : 22.3
RabbitMQ : 3.8.9.

**Reply by lukebakken** [reply]
Thanks. Please either provide a script or a _detailed_ set of steps I can follow to reproduce this issue.

**Reply by dfcarver** [reply]
Thank you.   Let me work on reproducing this.

**Reply by dfcarver** [reply]
Our client is in C#.   This is the same code that was used against the classic queues.  The code snippet is as follows:

[snippet.txt](https://github.com/rabbitmq/rabbitmq-server/files/7914632/snippet.txt)

**Reply by dfcarver** [reply]
We checked the log and we saw this message: 

2022-01-21 07:10:58.902 [info] <0.24831.33> accepting AMQP connection <0.24831.33> (10.26.26.40:35692 -> 10.26.26.23:5672)
2022-01-21 07:10:58.917 [info] <0.24831.33> connection <0.24831.33> (10.26.26.40:35692 -> 10.26.26.23:5672): user 'ntta' authenticated and granted access to vhost 'STAG_ADJ'
2022-01-21 07:10:58.933 [error] <0.24838.33> Channel error on connection <0.24831.33> (10.26.26.40:35692 -> 10.26.26.23:5672, vhost: 'STAG_ADJ', user: 'ntta'), channel 1:
operation queue.declare caused a channel exception precondition_failed: inequivalent arg 'x-queue-type' for queue 'ha.ADJ_Queue' in vhost 'STAG_ADJ': received none but current is the value 'quorum' of type 'longstr'
2022-01-21 07:11:00.539 [warning] <0.24831.33> closing AMQP connection <0.24831.33> (10.26.26.40:35692 -> 10.26.26.23:5672, vhost: 'STAG_ADJ', user: 'ntta'):
client unexpectedly closed TCP connection

**Reply by lukebakken** [reply]
```
precondition_failed: inequivalent arg 'x-queue-type' for queue 'ha.ADJ_Queue' in vhost 'STAG_ADJ': received none but current is the value 'quorum' of type 'longstr'
```

Ah, that's it. Your applications must be updated so that when `QueueDeclare` is called the correct `arguments:` are used (untested code, this is from memory):

```dotnet
var queueDeclareArguments = new Dictionary<string, object>
{
    { "x-queue-type", "quorum" }
}

channel.QueueDeclare(queue: PaymentProcessingConstants.QueueName, durable: true, exclusive: false, autoDelete: false, arguments: queueDeclareArguments);
```

**Reply by dfcarver** [reply]
That's it.  Thank you @lukebakken
